test/unit/GenericPredictorTest.cc

```diff
@@ -166,33 +231,83 @@ TEST_F(GenericPredictorTest, flush) {
   auto predictor = simeng::GenericPredictor();
   // Add some entries to the RAS
   auto prediction = predictor.predict(8, BranchType::SubroutineCall, 8);
-  EXPECT_TRUE(prediction.taken);
+  EXPECT_TRUE(prediction.isTaken);
   EXPECT_EQ(prediction.target, 16);
   prediction = predictor.predict(24, BranchType::SubroutineCall, 8);
-  EXPECT_TRUE(prediction.taken);
+  EXPECT_TRUE(prediction.isTaken);
   EXPECT_EQ(prediction.target, 32);
   prediction = predictor.predict(40, BranchType::SubroutineCall, 8);
-  EXPECT_TRUE(prediction.taken);
+  EXPECT_TRUE(prediction.isTaken);
   EXPECT_EQ(prediction.target, 48);
 
   // Start getting entries from RAS
   prediction = predictor.predict(52, BranchType::Return, 0);
-  EXPECT_TRUE(prediction.taken);
+  EXPECT_TRUE(prediction.isTaken);
   EXPECT_EQ(prediction.target, 44);
   prediction = predictor.predict(36, BranchType::Return, 0);
-  EXPECT_TRUE(prediction.taken);
+  EXPECT_TRUE(prediction.isTaken);
   EXPECT_EQ(prediction.target, 28);
 
   // Flush address
   predictor.flush(36);
 
   // Continue getting entries from RAS
   prediction = predictor.predict(20, BranchType::Return, 0);
-  EXPECT_TRUE(prediction.taken);
+  EXPECT_TRUE(prediction.isTaken);
   EXPECT_EQ(prediction.target, 28);
   prediction = predictor.predict(16, BranchType::Return, 0);
-  EXPECT_TRUE(prediction.taken);
+  EXPECT_TRUE(prediction.isTaken);
   EXPECT_EQ(prediction.target, 12);
 }
 
+// Test that update correctly corrects the speculatively updated global history
+TEST_F(GenericPredictorTest, speculativeGlobalHistory) {
+  simeng::config::SimInfo::addToConfig(
+      "{Branch-Predictor: {BTB-Tag-Bits: 6, Saturating-Count-Bits: 2, "
```

Comment:
Is this a valid config, there's no type information for the BP?

Reply:
Agreed, and updated.  Not sure why it was okay before

Reply:
Model config uses a default value of perceptron here